src/libraries/Common/src/Interop/Windows/Kernel32/Interop.SetConsoleCtrlHandler.cs

```diff
@@ -14,7 +14,12 @@ internal static partial class Kernel32
 
         internal delegate bool ConsoleCtrlHandlerRoutine(int controlType);
 
+#if DLLIMPORTGENERATOR_ENABLED
```

Comment:
Who is setting this define? I don't see it set anywhere.

Reply:
The DllImportGenerator package itself sets it:
https://github.com/dotnet/runtimelab/blob/feature/DllImportGenerator/DllImportGenerator/DllImportGenerator/Microsoft.Interop.DllImportGenerator.props#L20

Reply:
Is the DllImportGenerator going to be an "opt-in" feature? Or will it be built-in to a user's project by default when they target `net6.0+`?

If it is the latter, then this define constant doesn't seem to add much value, since every `net6.0+` project will get it by default.

Reply:
The current plan is for DllImportGenerator to be "opt-in" and for .NET 6.0 to be used internally only.

The point of the define is to enable a well-defined constant so we can provide a code fix that conditionally changes the code to use GeneratedDllImport when on .NET 6.0 and still use DllImport downlevel.

Reply:
At the moment, it is not going to be exposed to users for `net6.0` at all. We are just targeting runtime right now.

The define is an attempt to make it easier for multi-targeting scenarios - like this file that also gets used for non-`net6.0+`.